src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/JsonSerializerWrapper.Reflection.cs

```diff
@@ -881,5 +884,91 @@ private int ReadExactlyFromSource(byte[] buffer, int offset, int count)
+            }
+
+            public override async Task<object> DeserializeWrapper(string value, JsonTypeInfo jsonTypeInfo)
+            {
+                return await JsonSerializer.DeserializeAsync(new MemoryStream(Encoding.UTF8.GetBytes(value)), jsonTypeInfo);
+            }
+
+            public override async Task<object> DeserializeWrapper(string json, Type type, JsonSerializerContext context)
+            {
+                return await JsonSerializer.DeserializeAsync(new MemoryStream(Encoding.UTF8.GetBytes(json)), type, context);
+            }
+
+            public override async Task<string> SerializeWrapper(object value, Type inputType, JsonSerializerOptions options = null)
+            {
+                Pipe pipe = new Pipe();
+                await JsonSerializer.SerializeAsync(pipe.Writer, value, inputType, options);
+                ReadResult result = await pipe.Reader.ReadAsync();
+
+                string stringResult = Encoding.UTF8.GetString(result.Buffer.ToArray());
+                pipe.Reader.AdvanceTo(result.Buffer.End);
+                return stringResult;
+            }
+
+            public override async Task<string> SerializeWrapper<T>(T value, JsonSerializerOptions options = null)
+            {
+                Pipe pipe = new Pipe();
+                await JsonSerializer.SerializeAsync(pipe.Writer, value, options);
+                ReadResult result = await pipe.Reader.ReadAsync();
+
+                string stringResult = Encoding.UTF8.GetString(result.Buffer.ToArray());
+                pipe.Reader.AdvanceTo(result.Buffer.End);
+                return stringResult;
+            }
+
+            public override async Task<string> SerializeWrapper(object value, Type inputType, JsonSerializerContext context)
+            {
+                Pipe pipe = new Pipe();
+                await JsonSerializer.SerializeAsync(pipe.Writer, value, inputType, context);
+                ReadResult result = await pipe.Reader.ReadAsync();
+
+                string stringResult = Encoding.UTF8.GetString(result.Buffer.ToArray());
+                pipe.Reader.AdvanceTo(result.Buffer.End);
+                return stringResult;
+            }
+
+            public override async Task<string> SerializeWrapper<T>(T value, JsonTypeInfo<T> jsonTypeInfo)
+            {
+                Pipe pipe = new Pipe();
+                await JsonSerializer.SerializeAsync(pipe.Writer, value, jsonTypeInfo);
+                ReadResult result = await pipe.Reader.ReadAsync();
+
+                string stringResult = Encoding.UTF8.GetString(result.Buffer.ToArray());
+                pipe.Reader.AdvanceTo(result.Buffer.End);
+                return stringResult;
+            }
+
+            public override async Task<string> SerializeWrapper(object value, JsonTypeInfo jsonTypeInfo)
+            {
+                Pipe pipe = new Pipe();
```

Comment:
This file additionally defines the `AsyncStreamSerializerWithSmallBuffer` class which sets an artificially small `JsonSerializerOptions.DefaultBufferSize` to stress test the async implementation. Is there any chance we could do the same with a pipe? I foresee this would become essential once `PipeReader` support is added.